core/server/models/base/utils.js

```diff
@@ -67,4 +68,51 @@ tagUpdate = {
     }
 };
 
+trustedDomainUpdate = {
+    fetchCurrentClient: function fetchCurrentPost(ClientModel, id, options) {
+        return ClientModel.forge({id: id}).fetch(_.extend({}, options, {withRelated: ['trustedDomains']}));
+    },
+
+    fetchMatchingDomains: function fetchMatchingDomains(DomainModel, domainsToMatch, options) {
+        if (_.isEmpty(domainsToMatch)) {
+            return false;
+        }
+        return DomainModel.forge()
+            .query('whereIn', 'trusted_domain', _.pluck(domainsToMatch, 'trusted_domain')).fetchAll(options);
+    },
+
+    removeDomain: function removeDomain(DomainModel, client, domain, options) {
+        return function () {
+            return DomainModel.destroy({trusted_domain: domain.trusted_domain, client_id: client.id}, options);
+        };
+    },
+
+    createDomain: function createDomain(DomainModel, client, domain, options) {
+        return function () {
+            //return DomainModel.add({trusted_domain: domain.trusted_domain, client_id: client.id}, options);
+
+            return client.trustedDomains().create({trusted_domain: domain.trusted_domain}, options);
```

Comment:
I think you need to use the same pattern as here: https://github.com/TryGhost/Ghost/pull/6195/files#diff-8ca6b928bab45f8c504b120eba34344dR36